model/host/host.go

```diff
@@ -1778,6 +1788,57 @@ func (h *Host) SetInstanceType(instanceType string) error {
 	return nil
 }
 
+func AggregateSpawnhostData() (*SpawnHostUsage, error) {
+	res := []SpawnHostUsage{}
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			StartedByKey: bson.M{"$exists": true, "$ne": ""},
+			StatusKey:    bson.M{"$in": evergreen.UpHostStatus},
+		}},
+		{"$group": bson.M{
+			"_id":         nil,
+			"hosts":       bson.M{"$sum": 1},
+			"unexpirable": bson.M{"$sum": bson.M{"$cond": []interface{}{"$" + NoExpirationKey, 1, 0}}},
+			"users":       bson.M{"$addToSet": "$" + StartedByKey},
+		}},
+		{"$project": bson.M{
+			"_id":                     "0",
+			"total_hosts":             "$hosts",
+			"total_unexpirable_hosts": "$unexpirable",
+			"num_users_with_hosts":    bson.M{"$size": "$users"},
+		}},
+	}
+
+	if err := db.Aggregate(Collection, pipeline, &res); err != nil {
+		return nil, errors.Wrap(err, "error aggregating hosts")
+	}
+	if len(res) == 0 {
```

Comment:
can we assume that if there are no hosts there are no residual volumes?

Reply:
No we can't, because users may have detached volumes but never deleted them.